### PR TITLE
introduce span attributes keys table

### PR DESCRIPTION
**Pull request by makeavish**
This table will be used for cases when user writes a custom key

### CODE REVIEW

File: exporter/clickhousetracesexporter/migrations/000023_span_attributes_mv.up.sql [hunk 1]
```diff
@@ -7,7 +7,7 @@ CREATE TABLE IF NOT EXISTS signoz_traces.span_attributes ON CLUSTER cluster (
     float64TagValue Nullable(Float64) CODEC(ZSTD(1)),
     isColumn bool CODEC(ZSTD(1)),
 ) ENGINE ReplacingMergeTree
-ORDER BY (tagKey, tagType, dataType, stringTagValue, float64TagValue)
+ORDER BY (tagKey, tagType, dataType, stringTagValue, float64TagValue, isColumn)
```
Comment (srikanthccv) [inline exporter/clickhousetracesexporter/migrations/000023_span_attributes_mv.up.sql:10]: Why is this needed here?

File: exporter/clickhousetracesexporter/writer.go [hunk 2]
```diff
@@ -167,6 +170,13 @@ func (w *SpanWriter) writeBatch(batch []*Span) error {
 			return err
 		}
 	}
+	if w.attributeKeyTable != "" {
```
Comment (srikanthccv) [inline exporter/clickhousetracesexporter/writer.go:173]: The serial insertions, one at a time, will affect the overall throughput. It wouldn't have been an issue if the batch consisted only attributes because they are relatively small, but here it is iterating on the whole span batch and preparing the insertion.

File: exporter/clickhousetracesexporter/migrations/000024_span_attributes_keys.up.sql [hunk 3]
```diff
@@ -0,0 +1,10 @@
+CREATE TABLE IF NOT EXISTS signoz_traces.span_attributes_keys ON CLUSTER cluster (
+    tagKey LowCardinality(String) CODEC(ZSTD(1)),
+    tagType Enum('tag', 'resource') CODEC(ZSTD(1)),
+    dataType Enum('string', 'bool', 'float64') CODEC(ZSTD(1)),
+    isColumn bool CODEC(ZSTD(1)),
+) ENGINE ReplacingMergeTree
+ORDER BY (tagKey, tagType, dataType, isColumn);
```
Comment (srikanthccv) [inline exporter/clickhousetracesexporter/migrations/000024_span_attributes_keys.up.sql:7]: Does this need to keep the values forever? Can it have the same maximum TTL as the index table?

File: exporter/clickhousetracesexporter/migrations/000023_span_attributes_mv.up.sql (hunk 1 above)
Comment (makeavish) [reply exporter/clickhousetracesexporter/migrations/000023_span_attributes_mv.up.sql:10]: It's possible that there are two keys with the same name but one is column and the other is not, as we discussed we don't want to put restrictions on insertion of such keys and have to handle them on query builder (while querying)

File: exporter/clickhousetracesexporter/writer.go (hunk 2 above)
Comment (makeavish) [reply exporter/clickhousetracesexporter/writer.go:173]: We can merge the `writeTagKeyBatch` and `writeTagBatch` to reduce iterations.

File: exporter/clickhousetracesexporter/migrations/000024_span_attributes_keys.up.sql (hunk 3 above)
Comment (makeavish) [reply exporter/clickhousetracesexporter/migrations/000024_span_attributes_keys.up.sql:7]: But it's expected that keys won't be more than 10s of thousands, keys table of logs also doesn't have a TTL

File: exporter/clickhousetracesexporter/migrations/000023_span_attributes_mv.up.sql (hunk 1 above)
Comment (srikanthccv) [reply exporter/clickhousetracesexporter/migrations/000023_span_attributes_mv.up.sql:10]: I think this will cause issues with materialising the column in future.

**Review by srikanthccv — APPROVED** [review]
